### PR TITLE
WIP: Add visualization for a 2D structured grid of actors.

### PR DESCRIPTION
**Pull request by MarcCote**
This PR adds a new `vtkInteractorStyle` to visualize actors in a grid.

After a whole day of digging in the VTK source code, I finally managed to implement the new InteractorStyle the right way. Now @Garyfallidis do you have any idea of how I could unit test that?

### CODE REVIEW

**Comment by Garyfallidis** [comment]
I am at the CHUS today and cannot help much but what you could do is have a look at the other tests (e.g. test_fvtk_actor etc) and create a snapshot and then check the number of objects you see using analyze_snapshot and analyze_renderer. 

Apart from that we cannot test what happens when you drag something etc. But at least we can test that things come out in a grid.

**Comment by MarcCote** [comment]
Finally got the Container, GridLayout, the Interactor and the auto_orient (in dipy.viz.utils.py) working correctly. Still missing some examples and the tests though.

**Comment by Garyfallidis** [comment]
Add todo for tests! :dart:

**Comment by Garyfallidis** [comment]
:dancers:

File: dipy/viz/actor.py [hunk 1]
```diff
@@ -911,4 +1034,204 @@ def figure(pic, interpolation='nearest'):
...
+
+    def add_to_renderer(self, ren):
+        """ Adds the items of this container to a given renderer. """
+        for item in self.items:
+            if isinstance(item, Container):
+                item.add_to_renderer(ren)
+            else:
+                ren.add(item)
+
+    def GetBounds(self):
```
Comment (Garyfallidis) [inline dipy/viz/actor.py:1117]: For all our functions keep the current Dipy/Pythonic style.
get_bounds()

File: dipy/viz/actor.py [hunk 2]
```diff
@@ -911,4 +1034,204 @@ def figure(pic, interpolation='nearest'):
...
+            bounds[::2] = np.minimum(bounds[::2], item_bounds[::2])
+            bounds[1::2] = np.maximum(bounds[1::2], item_bounds[1::2])
+
+        # Add padding, if any.
+        bounds[::2] -= self.padding[::2]
+        bounds[1::2] += self.padding[1::2]
+
+        return tuple(bounds)
+
+    def GetVisibility(self):
```
Comment (Garyfallidis) [inline dipy/viz/actor.py:1134]: same

File: dipy/viz/actor.py [hunk 3]
```diff
@@ -911,4 +1034,204 @@ def figure(pic, interpolation='nearest'):
...
+        return self._visibility
+
+    def SetVisibility(self, visibility):
```
Comment (Garyfallidis) [inline dipy/viz/actor.py:1137]: okay please change everywhere

File: dipy/viz/actor.py [hunk 4]
```diff
@@ -911,4 +1034,204 @@ def figure(pic, interpolation='nearest'):
...
+                ren.add(item)
+
+    def GetBounds(self):
+        """ Get the bounds of the container. """
+        bounds = np.zeros(6)    # x1, x2, y1, y2, z1, z2
+        bounds[::2] = np.inf    # x1, y1, z1
+        bounds[1::2] = -np.inf  # x2, y2, z2
+
+        for item in self.items:
+            item_bounds = item.GetBounds()
```
Comment (MarcCote) [inline dipy/viz/actor.py:1124]: `item` can be either a `vtkActor` or a `Container`. This is why `GetBounds` must be written this way. If we change the name we'll need to put a `if` statement everywhere we use those functions! Do you prefer the latter?

File: dipy/viz/actor.py (hunk 1 above)
Comment (Garyfallidis) [inline dipy/viz/actor.py:1117]: okay got it now

**Pull request merged**